### ISSUE TITLE
`size_scattering` is not needed in `scattering1d`

### ISSUE THREAD

**Issue by lostanlen** [kymatio/kymatio#836]
as discussed on Discord with @MuawizChaudhary 

All our `Scattering1D` frontends have this piece of code in the forward pass

```
if self.vectorize:
    size_scattering = precompute_size_scattering(
        self.J, self.Q, self.T, max_order=self.max_order, detail=True)
else:
    size_scattering = 0
    
S = scattering1d(x, self.backend.pad, self.backend.unpad, self.backend, self.log2_T, self.psi1_f, self.psi2_f,
         self.phi_f, max_order=self.max_order, average=self.average, pad_left=self.pad_left,
         pad_right=self.pad_right, ind_start=self.ind_start, ind_end=self.ind_end,
         oversampling=self.oversampling,
         vectorize=self.vectorize,
         size_scattering=size_scattering,
         out_type=self.out_type)
```

But in fact we could measure `size_scattering` directly in the `scattering1d` routine, without this call to `precompute_size_scattering`. This should simplify our 1D codebase, by removing kwargs and boilerplate code.

**Comment by lostanlen** [comment]
Update: actually `size_scattering` is currently not being used by `scattering1d`. Getting the output size in advance is only needed for `ScatteringKeras1D.compute_output_shape`. So it's really a pure reduction in LOC count, without needing to write new code.

**Comment by lostanlen** [comment]
closed by #837
